### PR TITLE
[WIP] App object refactor

### PR DESCRIPTION
**Pull request by tractorcow**
Main issue: https://github.com/silverstripe/silverstripe-framework/issues/6681

See https://github.com/silverstripe/silverstripe-framework/pull/7037 for framework pull request

### CODE REVIEW

File: code/Model/RedirectorPage.php [hunk 1]
```diff
@@ -120,7 +121,9 @@ public function syncLinkTracking()
     {
         if ($this->RedirectionType == 'Internal') {
             if ($this->LinkToID) {
-                $this->HasBrokenLink = SiteTree::get()->byID($this->LinkToID) ? false : true;
+                $this->HasBrokenLink = Versioned::get_by_stage(SiteTree::class, Versioned::DRAFT)
```
Comment (chillu) [inline code/Model/RedirectorPage.php]: That seems unrelated to the change?

Comment (tractorcow) [inline code/Model/RedirectorPage.php]: The tests for this were marked skipped. I un-commented them and fixed them.
